src/lib/update-lock.ts

```diff
@@ -8,30 +7,20 @@ import * as Lock from 'rwlock';
 import * as constants from './constants';
 import {
 	ENOENT,
-	EEXIST,
+	LockfileExistsError,
 	UpdatesLockedError,
 	InternalInconsistencyError,
+	ChildProcessError,
 } from './errors';
-import { getPathOnHost, pathExistsOnHost } from './fs-utils';
+import { getPathOnHost, pathExistsOnHost, exec } from './fs-utils';
 import * as config from '../config';
+import * as lockfile from './lockfile';
+import log from './supervisor-console';
 
-type asyncLockFile = typeof lockFileLib & {
-	unlockAsync(path: string): Bluebird<void>;
-	lockAsync(path: string): Bluebird<void>;
-};
-const lockFile = Bluebird.promisifyAll(lockFileLib) as asyncLockFile;
-export type LockCallback = (
-	appId: number,
-	opts: { force: boolean },
-	fn: () => PromiseLike<void>,
-) => Bluebird<void>;
+const BASE_LOCK_DIR = '/tmp/balena-supervisor/services';
```

Comment:
Need to account for legacy lock paths, see https://www.balena.io/docs/learn/deploy/release-strategy/update-locking/#location-of-the-lockfile

Reply:
Depends, can those devices do a supervisor update?

Reply:
After discussing, the paths in that link are the lock paths from the user app POV, so they are not relevant here. The paths on host will be either `/tmp/balena-supervisor/services/$APP_ID/$SERVICE_NAME/*updates.lock`. `entry.sh` already symlinks `/tmp/resin-supervisor` to `/tmp/balena-supervisor`, if this older path exists.